internal/cli/cmd/create/credentials.go

```diff
@@ -0,0 +1,201 @@
+	cmdOpts.addFlags(cmd)
+
+	return cmd
+
+}
+
+// addFlags adds the flags for the get credentials options to the provided
+// command.
+func (o *createCredentialsOptions) addFlags(cmd *cobra.Command) {
+	o.PrintFlags.AddFlags(cmd)
+	option.Project(
+		cmd.Flags(), &o.Project, o.Project,
+		"The project in which to create credentials. If not set, the default project will be used.",
+	)
+	option.Git(cmd.Flags(), &o.Git, "Create credentials for a Git repository.")
+	option.Helm(cmd.Flags(), &o.Helm, "Create credentials for a Helm chart repository.")
+	option.Image(cmd.Flags(), &o.Image, "Create credentials for a container image repository.")
+	option.RepoURL(cmd.Flags(), &o.RepoURL, "URL of the repository the credentials are for.")
+	option.RepoURLPattern(
+		cmd.Flags(), &o.RepoURLPattern,
+		"Regular expression matching multiple repositories the credentials are for.",
+	)
+	option.Username(cmd.Flags(), &o.Username, "Username for the credentials.")
+	option.Password(cmd.Flags(), &o.Password, "Password for the credentials.")
+
+	cmd.MarkFlagsOneRequired(option.GitFlag, option.HelmFlag, option.ImageFlag)
+	cmd.MarkFlagsMutuallyExclusive(option.GitFlag, option.HelmFlag, option.ImageFlag)
+
+	cmd.MarkFlagsOneRequired(option.RepoURLFlag, option.RepoURLPatternFlag)
+	cmd.MarkFlagsMutuallyExclusive(option.RepoURLFlag, option.RepoURLPatternFlag)
+
+	if err := cmd.MarkFlagRequired(option.UsernameFlag); err != nil {
+		panic(errors.Wrapf(err, "could not mark %s flag as required", option.UsernameFlag))
+	}
+}
+
+// complete sets the options from the command arguments.
+func (o *createCredentialsOptions) complete(args []string) {
+	o.Name = args[0]
+}
+
+// validate performs validation of the options. If the options are invalid, an
+// error is returned.
+func (o *createCredentialsOptions) validate() error {
+	var errs []error
+	// While the flags are marked as required, a user could still provide an empty
+	// string. This is a check to ensure that the flags are not empty.
+	if o.Project == "" {
+		errs = append(errs, errors.New("project is required"))
+	}
+	if o.RepoURL == "" && o.RepoURLPattern == "" {
+		errs = append(
+			errs,
+			errors.New("either repo-url or repo-url-pattern is required"),
+		)
+	}
+	if o.Username == "" {
+		errs = append(errs, errors.New("username is required"))
+	}
+	return goerrors.Join(errs...)
+}
+
+// run creates the credentials in the project based on the options.
+func (o *createCredentialsOptions) run(ctx context.Context) error {
+	for {
+		if o.Password != "" {
+			break
+		}
+		prompt := &survey.Password{
+			Message: "Repository password",
+		}
+		if err := survey.AskOne(prompt, &o.Password); err != nil {
+			return err
+		}
+	}
+
+	kargoSvcCli, err := client.GetClientFromConfig(ctx, o.Config, o.Option)
+	if err != nil {
+		return errors.Wrap(err, "get client from config")
+	}
+
+	var repoType string
+	if o.Git {
+		repoType = "git"
+	} else if o.Helm {
+		repoType = "helm"
+	} else if o.Image {
+		repoType = "image"
```

Comment:
Any reason for not using the exported types from the `credentials` package instead of redefining strings?

Reply:
Fixed in e729af4fee2c4c64342a15513b1e5650aa847909